packages/flutter/lib/src/material/dropdown_menu.dart

```diff
@@ -584,12 +584,14 @@ class _DropdownMenuState<T> extends State<DropdownMenu<T>> {
       filteredEntries = widget.dropdownMenuEntries;
       buttonItemKeys = List<GlobalKey>.generate(filteredEntries.length, (int index) => GlobalKey());
       _menuHasEnabledItem = filteredEntries.any((DropdownMenuEntry<T> entry) => entry.enabled);
-      // If the text field content matches one of the new entries do not rematch the initialSelection.
-      final bool isCurrentSelectionValid = filteredEntries.any(
-        (DropdownMenuEntry<T> entry) => entry.label == _localTextEditingController?.text
-      );
-      if (!isCurrentSelectionValid) {
-        _localTextEditingController?.value = _initialTextEditingValue;
+      if (widget.initialSelection != null) {
+        // Rematch initialSelection if the text field content does not match one of the new entries.
+        final bool isCurrentSelectionValid = filteredEntries.any(
+          (DropdownMenuEntry<T> entry) => entry.label == _localTextEditingController?.text
+        );
+        if (!isCurrentSelectionValid) {
+          _localTextEditingController?.value = _initialTextEditingValue;
+        }
```

Comment:
> I think it is better to let the app control the behavior (by clearing explicitely the controller value) than forcing the reset as done since https://github.com/flutter/flutter/pull/155757. Not forcing the reset by default gives more flexibility.

I agree: not forcing the reset is probably the best way to go—or in other words, deleting this if-statement and its body entirely.

If the developer wants to trigger a `TextEditingController` change, the implementation should be pretty straightforward. The same cannot be said for when there's a non-null `initialSelection` and the developer wants to rebuild with a new entry list _without_ triggering a controller change.